runtime/jvmti/jvmtiClass.c

```diff
@@ -366,13 +366,25 @@ jvmtiGetClassSignature(jvmtiEnv* env,
 					signature[arity] = 'L';
 					memcpy(signature + arity + 1, J9UTF8_DATA(utf), utfLength);
 					signature[allocSize - 1] = ';';
+
+					if (J9ROMCLASS_IS_ANON_OR_HIDDEN(leafType->romClass)) {
+						int i = allocSize - 3;
+						for (; i >= 0; i--) {
+							/* hidden/anon class names are generated from ROMClassBuilder::handleAnonClassName() and have 0x<romaddress> appended at the end */
+							if ((ANON_CLASSNAME_CHARACTER_SEPARATOR == signature[i]) && ('0' == signature[i+1]) && ('x' == signature[i+2])) {
```

Comment:
Can you split the comparisons into 3 lines, i.e.:
```
if ((ANON_CLASSNAME_CHARACTER_SEPARATOR == signature[i]) 
    && ('0' == signature[i+1]) 
    && ('x' == signature[i+2])
) {
```

Reply:
updated in 841425c